### PR TITLE
fix: silent warning message & wrong exit status check

### PR DESCRIPTION
**Pull request by xiaoshihou514**
Just been testing with selene and found two issues.
- `on_failed` was silent again(process %s exited ... message did not show up), so I reverted a change.
- Finally I found that the exist status check was not behaving as expected as it exits with exit code 1 and an empty stderr(which should have been fine!) Guess I wasn't paying enough attention previously :sweat_smile:

### CODE REVIEW

**Comment by xiaoshihou514** [comment]
Maybe related to #77 , idk

File: lua/guard/lint.lua [hunk 1]
```diff
@@ -43,9 +43,11 @@ local debounce_timer = nil
 local function register_lint(ft, extra)
   api.nvim_create_autocmd('FileType', {
     pattern = ft,
+    group = 'Guard',
```
Comment (glepnir) [inline lua/guard/lint.lua:46]: use `api.nvim_create_augroup('Guard', {clear = false}) `?

File: lua/guard/lint.lua [hunk 2]
```diff
@@ -43,7 +43,7 @@ local debounce_timer = nil
 local function register_lint(ft, extra)
   api.nvim_create_autocmd('FileType', {
     pattern = ft,
-    group = 'Guard',
+    group = { 'Guard', { clear = true } },
```
Comment (glepnir) [inline lua/guard/lint.lua]: ?

File: lua/guard/spawn.lua [hunk 3]
```diff
@@ -67,12 +67,12 @@ local function spawn(opt)
       end
     end)
 
-    if exit_code == 0 and num_stderr_chunks == 0 then
-      coroutine.resume(co, table.concat(chunks))
-    else
+    if exit_code ~= 0 and num_stderr_chunks ~= 0 then
```
Comment (glepnir) [inline lua/guard/spawn.lua:70]: there should not be `or` ?

Comment (xiaoshihou514) [reply lua/guard/spawn.lua:70]: That makes it equivalent to the original code, where _either_ a non-zero exit code or non empty stderr indicates a lint failure. The problem is linters return non-zero exit code (for instance selene returns 1 when it found errors in given file) when the lint _succeeds_. So here _both_ need to be fulfilled before one could determine that the lint fails.

**Comment by xiaoshihou514** [comment]
> Thanks for this, was suspect on those changes I implemented - my bad. 

I should have done more testing! My laptop wasn't available at that timing :( It's definitely best to be more cautious next time, so as not to break someone's workflow.

File: lua/guard/spawn.lua (hunk 3 above)
Comment (barrett-ruth) [reply lua/guard/spawn.lua:70]: Is this change not the same as keeping the original code and replacing the `and` with an `or`?

Comment (xiaoshihou514) [reply lua/guard/spawn.lua:70]: > Is this change not the same as keeping the original code and replacing the `and` with an `or`?

It is, but I feel this way it's clearer what is being achieved

**Pull request merged**